src/lib/index.ts

```diff
@@ -66,12 +66,11 @@ class ServerlessIamPerFunctionPlugin {
     let length=0; //calculate the expected length. Sum the length of each part
     for (const part of name_parts) {
       if (part.Ref) {
-        length += part.Ref.length;
+        length += this.serverless.service.provider.region.length
```

Comment:
I am concerned with breaking something for someone that has a different ref. Let's use the region only when the ref is `AWS::Region`.
```suggestion
        if (part.Ref === 'AWS::Region') {
          length += this.serverless.service.provider.region.length;
        }
        else {  // this is not correct but for maintaining compatibility with previous logic we leave it until we can resolve other ref types
          length += part.Ref.length;
        }
```